comfyui_utils/comfy.py

```diff
@@ -88,33 +93,69 @@ async def _prompt_websocket(sess: PromptSession, callbacks: Callbacks) -> None:
-                    await callbacks.completed(queue_or_result, True)
+            if msg.type == aiohttp.WSMsgType.TEXT:
+                message = json.loads(msg.data)
+                # Handle prompt being started.
+                if message["type"] == "status":
+                    queue_or_result = await _get_queue_position_or_cached_result(sess)
+                    if isinstance(queue_or_result, int):
+                        await callbacks.queue_position(queue_or_result)
+                    else:
+                        await callbacks.completed(queue_or_result, True)
+                        break
+                # Handle a node being executed.
+                if message["type"] == "executing":
+                    if message["data"]["node"] is not None:
+                        node_id = int(message["data"]["node"])
+                        current_node = node_id
+                        await callbacks.in_progress(current_node, 0, 0)
+                # Handle completion of the request.
+                if message["type"] == "executed":
+                    assert message["data"]["prompt_id"] == sess.prompt_id
+                    await callbacks.completed(message["data"]["output"], False)
                     break
-            # Handle a node being executed.
-            if message["type"] == "executing":
-                if message["data"]["node"] is not None:
-                    node_id = int(message["data"]["node"])
-                    current_node = node_id
-                    await callbacks.in_progress(current_node, 0, 0)
-            # Handle completion of the request.
-            if message["type"] == "executed":
-                assert message["data"]["prompt_id"] == sess.prompt_id
-                await callbacks.completed(message["data"]["output"], False)
-                break
-            # Handle progress on a node.
-            if message["type"] == "progress":
-                progress = int(message["data"]["value"])
-                total = int(message["data"]["max"])
-                await callbacks.in_progress(current_node, progress, total)
-
+                # Handle progress on a node.
+                if message["type"] == "progress":
+                    progress = int(message["data"]["value"])
+                    total = int(message["data"]["max"])
+                    await callbacks.in_progress(current_node, progress, total)
+            elif msg.type == aiohttp.WSMsgType.BINARY:
+                image= await receive_image(msg.data)
+                if image is not None:
+                    await callbacks.image_received(image)
+            else:
+                logging.warning("Not text message, message type: %r", msg.type)            
+
+
+async def receive_image(image_data) -> Optional[Image.Image]:
+    '''
+    Reference : https://github.com/comfyanonymous/ComfyUI.git server.py function send_image
+    Rebuild an PIL Image from the data received on the websocket. Return None on any errors.
```

Comment:
nit: "a PIL image" + perhaps it's ok to skip the try and just let exception bubble up (ComfyAPI already throws exceptions)